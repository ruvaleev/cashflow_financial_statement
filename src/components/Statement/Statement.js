import React from  'react';

import '../../assets/stylesheets/index.css';
import { FormRow, FormSelectRow } from '../shared/formsElements';

function Statement({addIncome, incomes}) {
  return (
    <>
      <NewIncomeForm dispatch={addIncome}/>
      <div className='blackTitle'>Доходы</div>
      <TableRow firstTitle='Описание' secondTitle='Доход' className='tableHeader bordered'/>
      {incomes && incomes.map((income, i) => {
        return <TableRow key={i} firstTitle={income.title} secondTitle={income.value} className='bordered withoutTopBorder'/>
      })}
    </>
  )
}

function NewIncomeForm({dispatch}) {
  return (
    <form className='newIncomeForm bordered' onSubmit={(e) => {
      e.preventDefault();
      dispatch({
        title: e.target.title.value,
        value: e.target.value.value,
        isPassive: e.target.isPassive.value
      }),
      e.target.title.value = '',
      e.target.value.value = '';
    }}>
      <FormRow type='text' name='title' label='Статья дохода' required={true} className='bordered withoutTopBorder'/>
      <FormRow type='number' name='value' label='Сумма' className='bordered withoutTopBorder'/>
      <FormSelectRow
        name='isPassive'
        label='Доход пассивный?'
        className='bordered withoutTopBorder'
        options={[
          { value: true, title: 'Да' },
          { value: false, title: 'Нет' }
        ]}/>
      <button className='blackTitle blackButton'>Завести доход</button>
    </form>
  )
}

function TableRow({firstTitle, secondTitle, className}) {
  return (
    <div className={`tableRow ${className}`}>
      <div className='leftSide'>{firstTitle}</div>
      <div className='rightSide'>{secondTitle}</div>
    </div>
  )
}

export default Statement;
