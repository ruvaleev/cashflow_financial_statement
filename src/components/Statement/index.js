import { connect } from 'react-redux';

import Statement from './Statement';
import { addIncome } from '../../redux/slices/incomes';

const mapStateToProps = (state) => ({
  incomes: state.incomesReducer.incomes
})

const mapDispatchToProps = (dispatch) => ({
  addIncome: (data) => dispatch(addIncome(data))
})

export default connect(mapStateToProps, mapDispatchToProps)(Statement);
