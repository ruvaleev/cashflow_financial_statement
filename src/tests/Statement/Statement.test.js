import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';

import Store from '../shared/Store';
import Statement from '../../components/Statement';

const mockStore = configureStore([]);

describe('Statement', () => {
  let store;
  let component;
  
  beforeEach(() => {
    store = mockStore(Store());

    component = render(
      <Provider store={store}>
        <Statement/>
      </Provider>
    )
  });

  it('renders Statement correctly', () => {
    expect(component).toMatchSnapshot();
  });
});
