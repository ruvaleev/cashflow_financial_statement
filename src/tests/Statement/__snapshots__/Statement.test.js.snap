// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Statement renders Statement correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="controlPanel"
      >
        <div
          class="incomeForm"
        >
          <form
            class="bordered"
          >
            <div
              class="flex"
            >
              <label
                for="title"
              >
                Статья дохода
              </label>
              <input
                class="bordered withoutTopBorder"
                id="title"
                name="title"
                required=""
                type="text"
              />
            </div>
            <div
              class="flex"
            >
              <label
                for="value"
              >
                Сумма
              </label>
              <input
                class="bordered withoutTopBorder"
                id="value"
                name="value"
                type="number"
              />
            </div>
            <div>
              <label
                for="isPassive"
              >
                Доход пассивный?
              </label>
              <select
                class="bordered withoutTopBorder"
                id="isPassive"
                name="isPassive"
              >
                <option
                  value="true"
                >
                  Да
                </option>
                <option
                  value="false"
                >
                  Нет
                </option>
              </select>
            </div>
            <button
              class="blackTitle blackButton font-18px"
            >
              Завести доход
            </button>
          </form>
        </div>
        <div
          class="undoRedoPanel"
        >
          <button
            class="blackTitle blackButton font-16px noLeftMargin"
          >
            Отменить
          </button>
          <button
            class="blackTitle blackButton font-16px noRightMargin"
          >
            Повторить
          </button>
        </div>
        <div
          class="expenseForm"
        >
          <form
            class="bordered"
          >
            <div
              class="flex"
            >
              <label
                for="title"
              >
                Статья расхода
              </label>
              <input
                class="bordered withoutTopBorder"
                id="title"
                name="title"
                required=""
                type="text"
              />
            </div>
            <div
              class="flex"
            >
              <label
                for="value"
              >
                Сумма
              </label>
              <input
                class="bordered withoutTopBorder"
                id="value"
                name="value"
                type="number"
              />
            </div>
            <button
              class="blackTitle blackButton font-18px"
            >
              Завести расход
            </button>
          </form>
        </div>
        <div
          class="undoRedoPanel"
        >
          <button
            class="blackTitle blackButton font-16px noLeftMargin"
          >
            Отменить
          </button>
          <button
            class="blackTitle blackButton font-16px noRightMargin"
          >
            Повторить
          </button>
        </div>
        <button
          class="blackTitle blackButton font-16px"
        >
          Сохранить PDF
        </button>
      </div>
      <div
        id="statementDocumentContainer"
      >
        <div
          id="statementDocument"
        >
          <h1>
            Отчет о Доходах и Расходах
          </h1>
          <div
            class="blackTitle font-20px"
          >
            Доходы
          </div>
          <div
            class="tableContent flex"
          >
            <div
              class="leftPart"
            >
              <div
                class="flex tableRow tableHeader bordered"
              >
                <div
                  class="leftSide"
                >
                  Описание
                </div>
                <div
                  class="rightSide"
                >
                  Доход
                </div>
              </div>
              <div
                class="flex tableRow bordered withoutTopBorder"
              >
                <div
                  class="leftSide"
                >
                  зарплата
                </div>
                <div
                  class="rightSide"
                >
                  300
                </div>
              </div>
              <div
                class="flex tableRow bordered withoutTopBorder"
              >
                <div
                  class="leftSide"
                >
                  недвижимость
                </div>
                <div
                  class="rightSide"
                >
                  200
                </div>
              </div>
            </div>
            <div
              class="rightPart flex bordered"
            >
              <div>
                <div>
                  <strong>
                    Пассивный доход: 
                  </strong>
                  200
                </div>
                <div>
                  <strong>
                    Общий доход: 
                  </strong>
                  500
                </div>
              </div>
            </div>
          </div>
          <div
            class="blackTitle font-20px"
          >
            Расходы
          </div>
          <div
            class="tableContent flex"
          >
            <div
              class="leftPart"
            >
              <div
                class="flex tableRow tableHeader bordered"
              >
                <div
                  class="leftSide"
                >
                  Описание
                </div>
                <div
                  class="rightSide"
                >
                  Расход
                </div>
              </div>
              <div
                class="flex tableRow bordered withoutTopBorder"
              >
                <div
                  class="leftSide"
                >
                  ипотека
                </div>
                <div
                  class="rightSide"
                >
                  200
                </div>
              </div>
              <div
                class="flex tableRow bordered withoutTopBorder"
              >
                <div
                  class="leftSide"
                >
                  кредит за обучение
                </div>
                <div
                  class="rightSide"
                >
                  100
                </div>
              </div>
            </div>
            <div
              class="rightPart flex bordered"
            >
              <div>
                <strong>
                  Общий расход: 
                </strong>
                300
              </div>
            </div>
          </div>
          <div
            class="cashFlowResult font-20px"
          >
            <strong>
              Ежемесячный денежный поток: 
            </strong>
            200
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="controlPanel"
    >
      <div
        class="incomeForm"
      >
        <form
          class="bordered"
        >
          <div
            class="flex"
          >
            <label
              for="title"
            >
              Статья дохода
            </label>
            <input
              class="bordered withoutTopBorder"
              id="title"
              name="title"
              required=""
              type="text"
            />
          </div>
          <div
            class="flex"
          >
            <label
              for="value"
            >
              Сумма
            </label>
            <input
              class="bordered withoutTopBorder"
              id="value"
              name="value"
              type="number"
            />
          </div>
          <div>
            <label
              for="isPassive"
            >
              Доход пассивный?
            </label>
            <select
              class="bordered withoutTopBorder"
              id="isPassive"
              name="isPassive"
            >
              <option
                value="true"
              >
                Да
              </option>
              <option
                value="false"
              >
                Нет
              </option>
            </select>
          </div>
          <button
            class="blackTitle blackButton font-18px"
          >
            Завести доход
          </button>
        </form>
      </div>
      <div
        class="undoRedoPanel"
      >
        <button
          class="blackTitle blackButton font-16px noLeftMargin"
        >
          Отменить
        </button>
        <button
          class="blackTitle blackButton font-16px noRightMargin"
        >
          Повторить
        </button>
      </div>
      <div
        class="expenseForm"
      >
        <form
          class="bordered"
        >
          <div
            class="flex"
          >
            <label
              for="title"
            >
              Статья расхода
            </label>
            <input
              class="bordered withoutTopBorder"
              id="title"
              name="title"
              required=""
              type="text"
            />
          </div>
          <div
            class="flex"
          >
            <label
              for="value"
            >
              Сумма
            </label>
            <input
              class="bordered withoutTopBorder"
              id="value"
              name="value"
              type="number"
            />
          </div>
          <button
            class="blackTitle blackButton font-18px"
          >
            Завести расход
          </button>
        </form>
      </div>
      <div
        class="undoRedoPanel"
      >
        <button
          class="blackTitle blackButton font-16px noLeftMargin"
        >
          Отменить
        </button>
        <button
          class="blackTitle blackButton font-16px noRightMargin"
        >
          Повторить
        </button>
      </div>
      <button
        class="blackTitle blackButton font-16px"
      >
        Сохранить PDF
      </button>
    </div>
    <div
      id="statementDocumentContainer"
    >
      <div
        id="statementDocument"
      >
        <h1>
          Отчет о Доходах и Расходах
        </h1>
        <div
          class="blackTitle font-20px"
        >
          Доходы
        </div>
        <div
          class="tableContent flex"
        >
          <div
            class="leftPart"
          >
            <div
              class="flex tableRow tableHeader bordered"
            >
              <div
                class="leftSide"
              >
                Описание
              </div>
              <div
                class="rightSide"
              >
                Доход
              </div>
            </div>
            <div
              class="flex tableRow bordered withoutTopBorder"
            >
              <div
                class="leftSide"
              >
                зарплата
              </div>
              <div
                class="rightSide"
              >
                300
              </div>
            </div>
            <div
              class="flex tableRow bordered withoutTopBorder"
            >
              <div
                class="leftSide"
              >
                недвижимость
              </div>
              <div
                class="rightSide"
              >
                200
              </div>
            </div>
          </div>
          <div
            class="rightPart flex bordered"
          >
            <div>
              <div>
                <strong>
                  Пассивный доход: 
                </strong>
                200
              </div>
              <div>
                <strong>
                  Общий доход: 
                </strong>
                500
              </div>
            </div>
          </div>
        </div>
        <div
          class="blackTitle font-20px"
        >
          Расходы
        </div>
        <div
          class="tableContent flex"
        >
          <div
            class="leftPart"
          >
            <div
              class="flex tableRow tableHeader bordered"
            >
              <div
                class="leftSide"
              >
                Описание
              </div>
              <div
                class="rightSide"
              >
                Расход
              </div>
            </div>
            <div
              class="flex tableRow bordered withoutTopBorder"
            >
              <div
                class="leftSide"
              >
                ипотека
              </div>
              <div
                class="rightSide"
              >
                200
              </div>
            </div>
            <div
              class="flex tableRow bordered withoutTopBorder"
            >
              <div
                class="leftSide"
              >
                кредит за обучение
              </div>
              <div
                class="rightSide"
              >
                100
              </div>
            </div>
          </div>
          <div
            class="rightPart flex bordered"
          >
            <div>
              <strong>
                Общий расход: 
              </strong>
              300
            </div>
          </div>
        </div>
        <div
          class="cashFlowResult font-20px"
        >
          <strong>
            Ежемесячный денежный поток: 
          </strong>
          200
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
